drivers/modem/quectel-bg9x.c

```diff
@@ -0,0 +1,1141 @@
+/*
+ * Copyright (c) 2020 Analog Life LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT quectel_bg9x
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(modem_quectel_bg9x, CONFIG_MODEM_LOG_LEVEL);
+
+#include "quectel-bg9x.h"
+
+static struct k_thread	       modem_rx_thread;
+static struct k_work_q	       modem_workq;
```

Comment:
@bwasim Also I am not sure about the purpose of this workqueue. I see it's created and never consumed or no work/delayed work is posted to this queue. Is it consumed from modem layer or somewhere else?

Reply:
This is used to get the RSSI continuously during the Modem Operation. See modem_setup / modem_rssi_query_work for details.